x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/planner/QueryTranslator.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.eql.planner;
+
+import org.elasticsearch.xpack.eql.expression.function.scalar.string.CIDRMatch;
+import org.elasticsearch.xpack.ql.QlIllegalArgumentException;
+import org.elasticsearch.xpack.ql.expression.Expression;
+import org.elasticsearch.xpack.ql.expression.Expressions;
+import org.elasticsearch.xpack.ql.expression.FieldAttribute;
+import org.elasticsearch.xpack.ql.expression.function.scalar.ScalarFunction;
+import org.elasticsearch.xpack.ql.expression.predicate.logical.And;
+import org.elasticsearch.xpack.ql.expression.predicate.logical.Or;
+import org.elasticsearch.xpack.ql.planner.ExpressionTranslator;
+import org.elasticsearch.xpack.ql.planner.ExpressionTranslators;
+import org.elasticsearch.xpack.ql.planner.TranslatorHandler;
+import org.elasticsearch.xpack.ql.querydsl.query.Query;
+import org.elasticsearch.xpack.ql.querydsl.query.ScriptQuery;
+import org.elasticsearch.xpack.ql.querydsl.query.TermsQuery;
+import org.elasticsearch.xpack.ql.util.CollectionUtils;
+
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+
+import static org.elasticsearch.xpack.ql.planner.ExpressionTranslators.and;
+import static org.elasticsearch.xpack.ql.planner.ExpressionTranslators.or;
+
+final class QueryTranslator {
+
+    public static final List<ExpressionTranslator<?>> QUERY_TRANSLATORS = List.of(
+            new ExpressionTranslators.BinaryComparisons(),
+            new ExpressionTranslators.Ranges(),
+            new BinaryLogic(),
+            new ExpressionTranslators.Nots(),
+            new ExpressionTranslators.Likes(),
+            new ExpressionTranslators.InComparisons(),
+            new ExpressionTranslators.StringQueries(),
+            new ExpressionTranslators.Matches(),
+            new ExpressionTranslators.MultiMatches(),
+            new Scalars()
```

Comment:
preferred without static importing everything here, more visible QL translators vs EQL.

Reply:
A relevant aspect here: you might want to update your branch from master, to incorporate the `StartsWith` changes from https://github.com/elastic/elasticsearch/pull/56404. In that PR, the `Scalars` from QL was updated to have `startsWith` use a PrefixQuery whenever possible. And for EQL and its QueryTranslator that you added, it will probably need the `Scalars` from QL, as well.